silq/pulses/pulse_types.py

```diff
@@ -673,10 +673,11 @@ class MultiSinePulse(Pulse):
 
     Parameters:
         name: Pulse name
-        frequency: main Pulse frequency - !!! to be checked !!!
-        frequencies: list of pulse frequencies
-        phase: Pulse phase
+        frequency: main (local oscillator) Pulse frequency
```

Comment:
It's a nice addition being able to specify the carrier frequency in the pulse.
With a default value of `None`, do you expect that the carrier frequency is chosen by the interface that implements the pulse?

Reply:
We always specify this frequency (haven't tested with None value, but would expect error), and then interface treats this as the main frequency of the pulse and determines carrier frequency depending on the pulse sequence, e.g. if it's the only pulse in pulse sequence, then carrier frequency = this main frequency; if we have another pulse at another frequency, then interface will determine carrier based on those two main frequencies

Reply:
In that case, since `frequency` is a required argument you shouldn't give it the default value of `None`. I also agree with Serwan that it should be renamed to something more informative.

Reply:
What kind of default value then should I set? I copied this from other pulses - they all have None default value. I don't see the reason setting to something by default (magnetic field always varies), unless you mean I should add some kind of condition/demand on this frequency...
Concerning the renaming  - I didn't rename on purpose, since when microwave interface is checking pulses in pulse_sequence it works with "frequency" parameter of the pulse - I just wanted to be compatible with the interface. But, I suppose, of course, I could change interface appropriately for this pulse...

Reply:
corrected